scripts/cleanup_old_vms.py

```diff
@@ -68,143 +57,141 @@ def match(matchers, vm_name):
-            # Known conf issue :)
-            provider = get_mgmt(provider_key)
-        vm_list = provider.list_vm()
-
-        for vm_name in vm_list:
-            try:
-                if not match(matchers, vm_name):
-                    continue
-
-                if (isinstance(provider, VMWareSystem) and
-                        provider.vm_status(vm_name) == 'poweredOff'):
-                    hostname = provider.get_vm_host_name(vm_name)
-                    vm_config_datastore = provider.get_vm_config_files_path(vm_name)
-                    datastore_url = provider.get_vm_datastore_path(vm_name, vm_config_datastore)
-                    vm_creation_time = get_vm_config_modified_time(hostname, vm_name,
-                                                                   datastore_url, provider_key)
-                else:
-                    vm_creation_time = provider.vm_creation_time(vm_name)
-
-                if vm_creation_time is False:
-                    # This VM must have some problem, include in report even though we can't delete
-                    status = provider.vm_status(vm_name)
-                    deleted_vms_list.append([provider_key,
-                                             vm_name,
-                                             NULL,  # can't know age, failed getting creation date
-                                             status,
-                                             FAIL])
-                    raise Exception  # the except block message is accurate in this case
-
-                if vm_creation_time + delta < now:
-                    vm_delta = now - vm_creation_time
-                    with lock:
-                        vms_to_delete[provider_key].add((vm_name, vm_delta))
-            except Exception as e:
-                logger.error(e)
-                logger.error('Failed to get creation/boot time for {} on {}'.format(
-                    vm_name, provider_key))
-                continue
-
-        with lock:
-            print('{} finished processing for matches'.format(provider_key))
+        provider_mgmt = get_mgmt(provider_key)
+        vm_list = provider_mgmt.list_vm()
+
+        # don't really need to thread this, fast enough
+        text_matched_vms = [name for name in vm_list if match(matchers, name)]
+        non_text_matching = set(vm_list) - set(text_matched_vms)
+        print('{}: NOT matching text filters: {}'.format(provider_key, non_text_matching))
+        print('{}: Matching text filters: {}'.format(provider_key, text_matched_vms))
+
+        # MP for vm scanning, could be many VMs per provider
+        # match_queue is passed through and processed by this method's caller
+        non_time_match_queue = Queue()
+        vm_proc_list = [
+            Process(target=scan_vm,
+                    args=(provider_key, vm_name, delta, match_queue,
+                          non_time_match_queue, scan_failure_queue))
+            for vm_name in text_matched_vms]
```

Comment:
I'm torn about this in general, because a lot of places it does make sense (and we have in our codebase) to use comprehension on multiple lines.

Here I think you're right, and it will be more clear in a regular loop.